### ISSUE TITLE
aws-api + aws-auth-cognito instantiation failure after updating library versions

### ISSUE THREAD

**Issue by djweber** [aws-amplify/amplify-android#1606]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin - Coroutines

### Amplify Categories

Authentication, REST API

### Gradle script dependencies

<details>
Old:

```kotlin
object Amplify {
    private const val version = "1.17.5"

    const val api = "com.amplifyframework:aws-api:$version"
    const val cognito = "com.amplifyframework:aws-auth-cognito:$version"

    object Core {
        const val kotlin = "com.amplifyframework:core-kotlin:0.3.0"
    }
}
```
New:
```kotlin
object Amplify {
    private const val version = "1.31.0"

    const val api = "com.amplifyframework:aws-api:$version"
    const val cognito = "com.amplifyframework:aws-auth-cognito:$version"

    object Core {
        const val kotlin = "com.amplifyframework:core-kotlin:0.15.0"
    }
}
```
</details>

### Environment information

<details>

```
------------------------------------------------------------
Gradle 7.0.2
------------------------------------------------------------

Build time:   2021-05-14 12:02:31 UTC
Revision:     1ef1b260d39daacbf9357f9d8594a8a743e2152e

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.10 (JetBrains s.r.o 11.0.10+0-b96-7249189)
OS:           Windows 10 10.0 amd64

```

</details>

### Describe the bug

Hello,

I am currently using the Amplify SDKs for API Gateway and Cognito, with Kotlin coroutine support. The Android app is only using `AMAZON_COGNITO_USER_POOLS` for API authorization (we are not using identity pools), and was using an older version of these libs:

```kotlin
object Amplify {
    private const val version = "1.17.5"

    const val api = "com.amplifyframework:aws-api:$version"
    const val cognito = "com.amplifyframework:aws-auth-cognito:$version"

    object Core {
        const val kotlin = "com.amplifyframework:core-kotlin:0.3.0"
    }
}
```

We then updated to the latest versions for each:

```kotlin
object Amplify {
    private const val version = "1.31.0"

    const val api = "com.amplifyframework:aws-api:$version"
    const val cognito = "com.amplifyframework:aws-auth-cognito:$version"

    object Core {
        const val kotlin = "com.amplifyframework:core-kotlin:0.15.0"
    }
}
```

However, when we update these dependencies, the API Gateway plugin now refuses to initialize, giving the following error:

```
D/AWSMobileClient: Using the SignInProviderConfig from `awsconfiguration.json`.
D/AWSMobileClient: Inspecting user state details
E/[REDACTED]: Could not initialize Amplify
    ApiAuthException{message=Failed to retrieve auth token from Cognito provider., cause=ApiAuthException{message=You must be signed-in with Cognito Userpools to be able to use getTokens, cause=null, recoverySuggestion=Sorry, we don't have a suggested fix for this error yet.}, recoverySuggestion=Check the application logs for details.}
        at com.amplifyframework.api.aws.auth.ApiRequestDecoratorFactory.forAuthType(ApiRequestDecoratorFactory.java:127)
        at com.amplifyframework.api.aws.AWSApiPlugin.configure(AWSApiPlugin.java:158)
        at com.amplifyframework.core.category.Category.configure(Category.java:91)
        at com.amplifyframework.core.Amplify.configure(Amplify.java:156)
        at com.amplifyframework.core.Amplify.configure(Amplify.java:126)
        ...
     Caused by: ApiAuthException{message=You must be signed-in with Cognito Userpools to be able to use getTokens, cause=null, recoverySuggestion=Sorry, we don't have a suggested fix for this error yet.}
        at com.amplifyframework.api.aws.sigv4.DefaultCognitoUserPoolsAuthProvider.fetchToken(DefaultCognitoUserPoolsAuthProvider.java:98)
        at com.amplifyframework.api.aws.sigv4.DefaultCognitoUserPoolsAuthProvider.getLatestAuthToken(DefaultCognitoUserPoolsAuthProvider.java:104)
        at com.amplifyframework.api.aws.auth.ApiRequestDecoratorFactory.forAuthType(ApiRequestDecoratorFactory.java:125)
        at com.amplifyframework.api.aws.AWSApiPlugin.configure(AWSApiPlugin.java:158) 
        at com.amplifyframework.core.category.Category.configure(Category.java:91) 
        at com.amplifyframework.core.Amplify.configure(Amplify.java:156) 
        at com.amplifyframework.core.Amplify.configure(Amplify.java:126) 
        ...
W/CognitoUserSession: CognitoUserSession is not valid because idToken is null.
W/AWSMobileClient: signalTokensNotAvailable
```

I am able to sign in, but any subsequent API request fails:

```
Could not connect to platform: java.lang.IllegalStateException: Failed to get plugin because configuration previously failed.  Check for failures by logging any exceptions thrown by Amplify.configure().
    java.lang.IllegalStateException: Failed to get plugin because configuration previously failed.  Check for failures by logging any exceptions thrown by Amplify.configure().
        at com.amplifyframework.core.category.Category.getPluginIfConfiguredOrThrow(Category.java:225)
        at com.amplifyframework.core.category.Category.getSelectedPlugin(Category.java:215)
        at com.amplifyframework.api.ApiCategory.get(ApiCategory.java:114)
        at com.amplifyframework.kotlin.api.KotlinApiFacade.get(KotlinApiFacade.kt:126)
        at com.amplifyframework.kotlin.api.Rest$DefaultImpls.get$default(Rest.kt:31)
        at [REDACTED]
        at [REDACTED]
        at [REDACTED]
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

However, if I re-launch the app, the plugins are able to fetch the tokens and the app proceeds as normal. I have not changed any infrastructure on the AWS side of things. Downgrading back to the original versions fixes the issue.

### Reproduction steps (if applicable)

1. Create empty Android app with original dependencies
2. Configure AWS on Application onCreate:
   ```kotlin
      try {
            Amplify.addPlugin(AWSApiPlugin())
            Amplify.addPlugin(AWSCognitoAuthPlugin())
            Amplify.configure(applicationContext)
        } catch (exception: AmplifyException) {
            Log.e("Amplify", "Could not configure Amplify", exception)
        }
   ```
3. Observe exception when adding `AWSApiPlugin()`
4. Attempt to sign in with `Amplify.Auth.signIn(username, password)`
5. Observe successful sign in
6. Make API call with API Gateway:
    ```kotlin
    val options = RestOptions.builder()
        .addPath("/endpoint")
        .build()

    val response = Amplify.API.get(options)
    ```
7. Observe Exception:
    ```kotlin
     java.lang.IllegalStateException: Failed to get plugin because configuration previously failed.  Check for failures by logging any exceptions thrown by Amplify.configure()
    ```

### Code Snippet

```kotlin
private fun configureAws() {
    Timber.d("Initializing Amplify")

    try {
        Amplify.addPlugin(AWSApiPlugin())
        Amplify.addPlugin(AWSCognitoAuthPlugin())
        Amplify.configure(applicationContext)
        Timber.d("Initialized Amplify");
    } catch (exception: AmplifyException) {
        Timber.e(exception, "Could not initialize Amplify")
    }
}
```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### amplifyconfiguration.json

```json
{
  "auth": {
    "plugins": {
      "awsCognitoAuthPlugin": {
        "CognitoUserPool": {
          "Default": {
            "PoolId": "[REDACTED]",
            "AppClientId": "[REDACTED]",
            "AppClientSecret": "[REDACTED]",
            "Region": "us-east-1"
          }
        }
      }
    }
  },
  "api": {
    "plugins": {
      "awsAPIPlugin": {
        "[REDACTED]": {
          "endpointType": "REST",
          "endpoint": "[REDACTED]",
          "region": "us-east-1",
          "authorizationType": "AMAZON_COGNITO_USER_POOLS"
        }
      }
    }
  }
}
```

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

### Additional information and screenshots

The only change made here was bumping the lib versions. Nothing else. The closest issue I can find related to this issue is in the [amplify-flutter repository](https://github.com/aws-amplify/amplify-flutter/issues/913)

**Comment by eeatonaws** [comment]
Thank you for reporting this. We are looking into the issue and will provide any updates here.

**Comment by eeatonaws** [comment]
A PR with the fix for this issue has been released as part of Amplify Android version 1.31.2.
